src/hotspot/share/opto/memnode.cpp

```diff
@@ -2448,6 +2448,12 @@ const Type* LoadNode::klass_value_common(PhaseGVN* phase) const {
     }
   }
 
+  if (tkls != nullptr && !UseSecondarySuperCache
```

Comment:
May be add comment what are you doing here.
Is it switching off **current** implementation of secondary super cache optimization or both: current and new?

Reply:
All it does is switch of the **current** implementation, and only in C2.